model/commitqueue/db.go

```diff
@@ -71,6 +71,26 @@ func add(id string, queue []CommitQueueItem, item CommitQueueItem) error {
 	return err
 }
 
+// assume that the first element in the array is being processed, so move after that
+func addToFront(id string, queue []CommitQueueItem, item CommitQueueItem) error {
+	err := updateOne(
+		bson.M{
+			IdKey: id,
+		},
+		bson.M{"$push": bson.M{
+			QueueKey: bson.M{
+				"$each":     []CommitQueueItem{item},
+				"$position": 1,
```

Comment:
in the case where `processing == false` would it work to put it at the front?

Reply:
I want to avoid a race where our in-memory queue has processing == false as the first element, but by the time we've inserted that isn't true anymore. I don't have stats on this but I imagine items aren't at the top of the queue with processing == false for long so I was thinking it's not worth it.